### The results are stored in the .pkl located in the wandb run directory. We need to undump this to look into the individual entropies of each semantic class

In [1]:
# Fill in the wandb_id assigned to your demo run!

wandb_id = 'd01zakoy'
if wandb_id == 'YOUR_ID':
    raise ValueError('Need to provide wandb_id of demo run!')

In [2]:
%load_ext autoreload
%autoreload 2

import os
import json
import wandb
import pandas as pd
from matplotlib import pyplot as plt

In [3]:
# Helper Functions
def restore_file(wandb_id, filename='wandb-summary.json'):
    files_dir = 'notebooks/restored_files'    
    os.system(f'mkdir -p {files_dir}')

    api = wandb.Api()
    run = api.run(f'semantic_uncertainty/{wandb_id}')

    path = f'{files_dir}/{filename}'
    os.system(f'rm -rf {path}')
    run.file(filename).download(root=files_dir, replace=True, exist_ok=False)
    with open(path, 'r') as f:
        out = json.load(f)
    return out

### come back and implement this later. For now, manually download uncertainty_measures.pkl from the wandb run instance and dissect it.

In [ ]:
results = restore_file(wandb_id, 'uncertainty_measures.pkl')

In [10]:
import pickle

# Load the contents of a pkl file into a dictionary
with open('uncertainty_measures.pkl', 'rb') as file:
    data = pickle.load(file)

# Print the loaded dictionary
print(type(data))

<class 'dict'>


In [ ]:
len(data['uncertainty_measures']['cluster_assignment_entropy'])

399

In [17]:
len(data['uncertainty_measures']['regular_entropy'])

399

### For some reason, we see only 399 entropies, while there were 400 data points sampled. Where is the missing one? Check with other pkl files too.